src/managers/queue-manager.js

```diff
@@ -75,6 +77,18 @@ async function handleMessage(message) {
 }
 
 export async function pullMessagesFromQueue() {
+  if (settings.hasActiveSSEConnection()) {
+    await checkMessageQueue();
+  } else {
+    if (settings.useSSE()) {
+      await setupSSEQueueListener();
+    } else {
+      await checkQueueThroughPolling();
+    }
+  }
```

Comment:
👀 Readability nitpick: We could reduce nesting here with guard clauses.

```suggestion
  if (settings.hasActiveSSEConnection()) {
    await checkMessageQueue();
    return;
  }
  
  if (settings.useSSE()) {
    await setupSSEQueueListener();
    return;
  }
  
  await checkQueueThroughPolling();
```